tests/spi/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! This is a primitive SPI test.
+//!
+//! It performs different reads and writes in different modes. The precise results depend on
+//! the voltage on the MISO pin, which is usually indeterminate.
+#![no_std]
+
+use embedded_hal::spi::SpiDevice;
+
+use riot_wrappers::println;
+use riot_wrappers::riot_main;
+
+riot_main!(main);
+
+fn main() {
+    // SPI device and CS pin that the test may drive.
+    //
+    // Boards should only be added if the attached peripherals are safe to use, no matter what gets
+    // written there.
+    let (spi_num, cs_num) = match riot_wrappers::BOARD {
+        "particle-xenon" => (0, (0, 31)),
+        _ => panic!("No "),
+    };
+
+    let cs = riot_wrappers::gpio::GPIO::from_port_and_pin(cs_num.0, cs_num.1).unwrap();
+    let spi = riot_wrappers::spi::for_embedded_hal_1::SPIBus::from_number(spi_num)
```

Comment:
```suggestion
    let cs: riot_wrappers::gpio::GPIO = riot_wrappers::gpio::GPIO::from_port_and_pin(cs_num.0, cs_num.1).unwrap();
    let spi: riot_wrappers::spi::SPIBus = riot_wrappers::spi::for_embedded_hal_1::SPIBus::from_number(spi_num)
```
Might be worth it, as tests are also examples how to use stuff.

Reply:
The latter is *not* the type; that would be `riot_wrappers::spi::for_embedded_hal_1::SPIBus`riot_wrappers::spi::for_embedded_hal_1::SPIBus`. (The v0.2 and v1 implementations are completely disjunct).

IMO, having the type in the declaration makes a lot of sense for clarity if
* the expression after it is not a constructor (but here it is), and especially when
* using `let x: u32 = something.into();` (where the type is inferred and otherwise you'd need a mouthful to express it)